utils/validators/validators_test.go

```diff
@@ -0,0 +1,146 @@
+			nil,
+			[]string{
+				`unknown environment variable "UNKNOWN_VAR=VAL"`,
+				`unknown environment variable "ANOTHER_UNKNOWN_VAR=VAL"`,
+			},
+		},
+		{
+			"Default env vars",
+			args{[]string{
+				"PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin",
+				"HOSTNAME=host",
+				"TERM=xterm-256color",
+				"HOME=/home/user/",
+			}},
+			map[string]string{
+				"HOME":     "/home/user/",
+				"HOSTNAME": "host",
+				"PATH":     "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin",
+				"TERM":     "xterm-256color",
+			},
+			nil,
+			nil,
+		},
+		{
+			"Invalid env variables values",
+			args{[]string{
+				"DISABLE_UPDATES=5",
+				"DISABLE_TELEMETRY=X",
+				"METRICS_RESOLUTION=5f",
+				"METRICS_RESOLUTION_HR=s5",
+				"METRICS_RESOLUTION_LR=1hour",
+				"DATA_RETENTION=keep one week",
+			}},
+			map[string]string{},
+			[]error{
+				errors.New(`invalid environment variable "DISABLE_UPDATES=5"`),
+				errors.New(`invalid environment variable "DISABLE_TELEMETRY=X"`),
+				errors.New(`invalid environment variable "METRICS_RESOLUTION=5f"`),
+				errors.New(`invalid environment variable "METRICS_RESOLUTION_HR=s5"`),
+				errors.New(`invalid environment variable "METRICS_RESOLUTION_LR=1hour"`),
+				errors.New(`invalid environment variable "DATA_RETENTION=keep one week"`),
+			},
+			nil,
+		},
+		{
+			"Grafana env vars",
+			args{[]string{
+				`GF_AUTH_GENERIC_OAUTH_ALLOWED_DOMAINS='example.com'`,
+				`GF_AUTH_GENERIC_OAUTH_ENABLED='true'`,
+				`GF_PATHS_CONFIG="/etc/grafana/grafana.ini"`,
+				`GF_PATHS_DATA="/var/lib/grafana"`,
+				`GF_PATHS_HOME="/usr/share/grafana"`,
+				`GF_PATHS_LOGS="/var/log/grafana"`,
+				`GF_PATHS_PLUGINS="/var/lib/grafana/plugins"`,
+				`GF_PATHS_PROVISIONING="/etc/grafana/provisioning"`,
+			}},
+			map[string]string{
+				"GF_AUTH_GENERIC_OAUTH_ALLOWED_DOMAINS": "'example.com'",
+				"GF_AUTH_GENERIC_OAUTH_ENABLED":         "'true'",
+				"GF_PATHS_CONFIG":                       `"/etc/grafana/grafana.ini"`,
+				"GF_PATHS_DATA":                         `"/var/lib/grafana"`,
+				"GF_PATHS_HOME":                         `"/usr/share/grafana"`,
+				"GF_PATHS_LOGS":                         `"/var/log/grafana"`,
+				"GF_PATHS_PLUGINS":                      `"/var/lib/grafana/plugins"`,
+				"GF_PATHS_PROVISIONING":                 `"/etc/grafana/provisioning"`,
+			},
+			nil,
+			nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotEnvVars, gotErrs, gotWarns := EnvVarValidator(tt.args.env)
+			if !reflect.DeepEqual(gotEnvVars, tt.wantEnvVars) {
+				t.Errorf("EnvVarValidator() gotEnvVars = %v, want %v", gotEnvVars, tt.wantEnvVars)
+			}
+			if !reflect.DeepEqual(gotErrs, tt.wantErrs) {
+				t.Errorf("EnvVarValidator() gotErrs = %v, want %v", gotErrs, tt.wantErrs)
+			}
+			if !reflect.DeepEqual(gotWarns, tt.wantWarns) {
+				t.Errorf("EnvVarValidator() gotWarns = %v, want %v", gotWarns, tt.wantWarns)
+			}
```

Comment:
`assert.Equal`, etc.

Please do not use that template when it does not match all other test files in this repo.